bentoml/tensorflow.py

```diff
@@ -433,22 +436,40 @@ def _setup(self) -> None:  # type: ignore[override]
         )
         self._model = load(self.name, model_store=self._model_store)
         if not TF2:
-            self._predict_fn = self._model.signatures["serving_default"]
+            raw_predict_fn = self._model.signatures["serving_default"]
         else:
-            self._predict_fn = getattr(self._model, self._predict_fn_name)
+            raw_predict_fn = getattr(self._model, self._predict_fn_name)
+        self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
 
     # pylint: disable=arguments-differ
     def _run_batch(  # type: ignore[override]
-        self, input_data: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor]
+        self,
+        *args: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor],
+        **kwargs: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor],
     ) -> np.ndarray:
-        with self._device:
```

Comment:
`tf.device(device_id)` cannot be cached because it returns a context manager and according to https://github.com/python/cpython/blob/main/Lib/contextlib.py#L135 the parameters of context will be deleted once entered the context.

Hence every time we call `_run_batch`, we need  to call `tf.device(device_id)`, or else the runner can only run once.

@aarnphm

Reply:
noted